### DISCUSSION TITLE
help...newbie got a "failed to set-up check logs" the log has abunch of "errors"

### DISCUSSION THREAD

**Discussion by spaulus69** [SecKatie/ha-wyzeapi#246] [Q&A]
Error while setting up wyzeapi platform for light
3:12:10 PM – (ERROR) Light (custom integration)
Error while setting up wyzeapi platform for lock
3:12:10 PM – (ERROR) Lock (custom integration)
Error while setting up wyzeapi platform for climate
3:12:10 PM – (ERROR) Climate (custom integration)
Error while setting up wyzeapi platform for alarm_control_panel
3:12:10 PM – (ERROR) Alarm control panel (custom integration)
Error while setting up wyzeapi platform for switch
3:12:10 PM – (ERROR) Switch (custom integration)
Error setting up entry Wyze for wyzeapi
3:12:10 PM – (ERROR) Wyze Home Assistant Integration (custom integration)
Error doing job: Unclosed client session
3:12:09 PM – (ERROR) runner.py - message first occurred at 3:12:09 PM and shows up 2 times

**Comment by JoeSchubert** [comment]
I know this is pretty old and it probably would've been an issue vice discussion....

I'm guessing that you didn't install through HACS as recommended. You'll need to manually install wyzeapy from pypi via pip if you're going to manually install the integration.

**Reply by tggman** [reply]
So I have a question. I've had no issues but... I just download the ha-wyzeapi.zip file And copy the contents of the custom_components folder it into my HA installation. Again I've had no issues that I know of but do I need to (somehow) update my version of the wyzeapi or is it done automatically or not necessary?

If I need to update it, how do I do that? I'm running supervised HAC in HassOS. Obviously I can use an SSH terminal but it doesn't support pip

I really have zero interest in using HACS since it is yet another path into my network and I think we both know the history on what occurred previously...

**Reply by JoeSchubert** [reply]
You're going to have problems if you can't use pip regardless of what you do, since that's how home assistant is installing the dependencies anyway. 

You could extract the files from wyzeapi's release zips directly into the python packages path. But again, you're going to constantly run into problems.

**Reply by tggman** [reply]
After a bit of research, I **believe** the manual installation method (i.e. not using HACS) should be okay as long as the requirements key in manifest.json file is kept up to date.

Here's what the developer docs say:
https://developers.home-assistant.io/docs/creating_integration_manifest/#requirements

and here is the manifest.json file:
{
  "domain": "wyzeapi",
  "version": "2021.10.2b1",
  "name": "Wyze",
  "config_flow": true,
  "iot_class": "cloud_polling",
  "documentation": "https://github.com/JoshuaMulliken/ha-wyzeapi#readme",
  "issue_tracker": "https://github.com/JoshuaMulliken/ha-wyzeapi/issues",
  "requirements": [
    "wyzeapy==0.5.0"
  ],
  "codeowners": [
    "@JoshuaMulliken"
  ]
}

Speaking of  version control/info, it sure would be nice if ha-wyzeapi either created a separate entity OR added attributes that contain the wyzeapi AND the ha-wyzeapi versions currently in use.

**Reply by JoeSchubert** [reply]
I don't know. During my testing, home assistant only pulled the required packages in the manifest during the initial installation, if you overwrite the files it did not.

Also, when have the requirements not been kept up to date (barring an accidental mismatch here or there)?

Why would we create entities for that? The recommended, (and only supported installation method) handle that already and let you know when there's an update. *Edit* just to clarify, I'm asking what the perceived benefit here is.

I imagine that Josh would entertain a pull request adding them though, if you want to want to put in the effort to add them for your usage.

**Reply by tggman** [reply]
Edit just to clarify, I'm asking what the perceived benefit here is?  Knowing the version of any piece of software is key from a maintenance standpoint - which is why every piece of software out there exposes it.

Here's the bottom line and then I'll stop. Maximizing the security of my Home Assistant installation and my LAN is essential. Every single add-on and/or integration is a potential path into my install - and HACS is no exception. Minimizing what I install on every single computing device I own is my best practice - a practice that has served me well for decades (e.g. see https://www.home-assistant.io/blog/2021/01/23/security-disclosure2/).  

So to be clear, I'd rather uninstall/reinstall ha-wyzeapi as required vs installing yet another integration that has minimal benefit (to me) - and knowing the versions of both ha-wyzeapi and wyzeapi will allow me to setup an automation that will let know when it's required vs using HACS.

**Reply by SecKatie** [reply]
Here are my 2 cents. I think that a sensor that has the version would be valuable in some cases but I don't have any interest in implementing that myself.

Installing outside of HACS is not something that I am going to "officially support" because it is not how I run my addons but I would accept PRs to add instructions to the README (explicitly stating that people should not file issues regarding that method but can join the discussion forums to discuss) or even code changes to make that installation method work better.

**Reply by JoeSchubert** [reply]
> Edit just to clarify, I'm asking what the perceived benefit here is? Knowing the version of any piece of software is key from a maintenance standpoint - which is why every piece of software out there exposes it.
> 

Not trying to insulting or anything here... But again, HACS provides this information already. Your usecase is outside of the supported configuration/installation method. Have you run into many projects out there that dedicate development time or effort toward unsupported configurations?

I'd be willing to bet that with fairly minimal effort you could include a sensor via yaml that would read the version from the manifest for you.

**Comment by tggman** [comment]
Obviously (?) grabbing the version number from the manifest file would be trivial. At least for me... You're missing the point but I'm done talking about this here at least.
